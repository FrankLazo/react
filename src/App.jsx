// import BreakingBadApp from './luis-cabrera/BreakingBadApp';
// import ContextApp from './luis-cabrera/ContextApp';
// import CustomHooksApp from './luis-cabrera/CustomHooksApp';
// import DogsApp from './luis-cabrera/DogsApp';
// import GlobalStateApp from './luis-cabrera/GlobalStateApp';
// import MemorizeApp from './luis-cabrera/MemorizeApp';
// import ModalApp from './luis-cabrera/ModalApp';
// import RouterApp from "./luis-cabrera/RouterApp";
// import TodoApp from './luis-cabrera/TodoApp';
// import UseEffectApp from './luis-cabrera/UseEffectApp';
// import UseReducerApp from './luis-cabrera/UseReducerApp';
// import UseRefApp from './luis-cabrera/UseRefApp';
// import UseStateApp from './luis-cabrera/UseStateApp';

import GifExpertApp from './fernando-herrera/04-gif-expert-app/GifExpertApp';

function App()
{
	return (
		<div className="App">
			{/* <TodoApp /> */}
			{/* <BreakingBadApp /> */}
			{/* <DogsApp /> */}
			{/* <UseStateApp /> */}
			{/* <UseEffectApp /> */}
			{/* <UseRefApp /> */}
			{/* <MemorizeApp /> */}
			{/* <CustomHooksApp /> */}
			{/* <ModalApp /> */}
			{/* <ContextApp /> */}
			{/* <UseReducerApp /> */}
			{/* <GlobalStateApp /> */}
			{/* <RouterApp /> */}

			<GifExpertApp />
		</div>
	);
}

export default App;
