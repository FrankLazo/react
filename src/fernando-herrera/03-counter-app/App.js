import PropTypes from 'prop-types';

const App = ({ welcome, description }) => {
	return (
		<>
			<h1>{ welcome }</h1>
			<p>{ description }</p>
		</>
	);
}

App.propTypes = {
  	description: PropTypes.string.isRequired,
}

App.defaultProps = {
  	description: 'My first application',
}

export default App;
