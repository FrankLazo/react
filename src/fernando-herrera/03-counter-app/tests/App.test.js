// import { render } from "@testing-library/react";
import { shallow } from 'enzyme';
import App from "../App";

describe('App', () => {
    // test('should be Hello World!', () => {
    //     const welcome = 'Hello World!';
    //     const { getByText } = render(<App welcome={ welcome } />);

    //     expect(getByText(welcome)).toBeInTheDocument();
    // });

	// Enzyme

    test('should be <App />', () => {
        const welcome = 'Hello World!';
		const wrapper = shallow(<App welcome={ welcome } />);
		expect(wrapper).toMatchSnapshot();
    });

	test('should show description by props', () => {
		const welcome = 'Hello World!';
		const description = 'My description';

		const wrapper = shallow(
			<App welcome={ welcome } description={ description } />
		);

		const paragraphText = wrapper.find('p').text();
		expect(paragraphText).toBe(description);
	});
});
