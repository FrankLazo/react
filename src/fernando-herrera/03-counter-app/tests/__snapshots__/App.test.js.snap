// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be <App /> 1`] = `
<Fragment>
  <h1>
    Hello World!
  </h1>
  <p>
    My first application
  </p>
</Fragment>
`;

exports[`App should be <CounterApp /> 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <div
    className="controls"
  >
    <button
      onClick={[Function]}
    >
      +1
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      -1
    </button>
  </div>
</Fragment>
`;
